### ISSUE TITLE
v6.1.1 hangs when stdin is discarded

### ISSUE THREAD

**Issue by timdp** [sindresorhus/ora#211]
With v6.1.1, this no longer exits:

```js
import ora from 'ora'
const spinner = ora('test').start()
setTimeout(() => spinner.succeed(), 1000)
```

Changing the arg to `{ text: 'test', discardStdin: false }` fixes it, which makes sense, as #210 introduced a change to the code that handles stdin.

I'm observing this on Node v14, v16, and v18.

**Comment by sindresorhus** [comment]
// @clavin

**Comment by clavin** [comment]
Yup, can repro this. This is one unfortunate way to learn that node will avoid exiting when you call resume on stdin. Apologies for breaking this! 🙇

I'm investigating if there's a possible solution for the original issue using another stream as a proxy; alternatively, I might have to contribute upstream to node to make this sort of functionality possible if it is not already (or learn why it isn't).

For right now it would be best to just revert #210. Unfortunately that might also mean #209 still is an open issue.
